### DISCUSSION TITLE
DOCKER: Missing a temporary folder in the Data-Importer V 0.8.0

### DISCUSSION THREAD

**Discussion by Offertorium** [firefly-iii/firefly-iii#5761] [Firefly III] [answered]
Hi,

if I want to use the data importer I get the following error: "Missing a temporary folder. "And the log says: "PHP Warning: File upload error - unable to create a temporary file in Unknown on line 0, referer: [Client-IP]" However, I was already able to upload files. After restarting the container, this is no longer possible. In which directory does he want to write the files? The Folder: - firefly_iii_upload:/var/www/html/storage/upload on the host machine is writable for the User. 

Thanks and Bye

My Environment:
Debug information generated at 2022-02-10 23:37:29 Europe/Berlin for Firefly III version **5.6.14**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.6.14 |
| Firefly III API | 1.5.5 |
| Build | 96, 05-02-2022 10:01:49 CET |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| Layout | v1 |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | de_DE |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| DB version | 18 (exp. 18) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2022-02-01 00:00:00 |
| Session end | 2022-02-28 23:59:59 |
| Session first | 2020-12-08 00:00:00 |
| User ID | 1 |
| User language | de_DE |
| User locale | de_DE |
| Attempt at "de_DE.utf8" | 'de_DE.utf8' |
| Attempt at "de_DE.UTF-8" | 'de_DE.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0 |

**Comment by JC5** [comment]
Mmm, I'll check it out. Haven't heard this yet from other users.

**Comment by JC5** [comment]
I'm not seeing this. Can you tell me more about your import process?

**Comment by Offertorium** [comment, marked as answer]
Hi,
thanks for your support. 
I upload the files regularly as CSV via the web interface. This has also worked a few times. 

I tested something around and was able to solve the problem. 
I copied the php.ini from the Data_Importer container: 
`docker cp Container-ID:/usr/local/etc/php/php.ini .`

In the php.ini i set the variable: 
`upload_tmp_dir = /tmp` 

and copied the php.ini back to the container: 
`docker cp php.ini Container-ID:/usr/local/etc/php/ `

Login in to the Container:
`docker exec -it Container-ID bash`

In the Container:
`chmod 777  /tmp `
and Apache restarted: 
`/etc/init.d/apache2 restart` 

_I have a suggestion, put vi in the container ;)_

After that, I was able to upload and import files again. I tried this with a few test data. I don't know if this is the correct and long-term solution? Also I hope that the google translated text is understandable. 

Thank you for your support.

**Reply by JC5** [reply]
I cannot reproduce the problem at all. It works fine for me. This will work until a new version is out. Then you have to do it again.

But I don't know the cause, sorry.
